sycl/include/sycl/ext/oneapi/reduction.hpp

```diff
@@ -368,52 +748,23 @@ template <typename T> struct AreAllButLastReductions<T> {
 
 /// This class encapsulates the reduction variable/accessor,
 /// the reduction operator and an optional operator identity.
-template <typename T, class BinaryOperation, int Dims, bool IsUSM,
-          access::placeholder IsPlaceholder = access::placeholder::false_t>
-class reduction_impl : private reduction_impl_base {
-public:
-  using reducer_type = reducer<T, BinaryOperation>;
-  using result_type = T;
-  using binary_operation = BinaryOperation;
-  using rw_accessor_type =
-      accessor<T, Dims, access::mode::read_write, access::target::device,
-               IsPlaceholder, ext::oneapi::accessor_property_list<>>;
-  using dw_accessor_type =
-      accessor<T, Dims, access::mode::discard_write, access::target::device,
-               IsPlaceholder, ext::oneapi::accessor_property_list<>>;
-  static constexpr int accessor_dim = Dims;
-  static constexpr int buffer_dim = (Dims == 0) ? 1 : Dims;
-
-  static constexpr bool has_atomic_add_float64 =
-      IsReduOptForAtomic64Add<T, BinaryOperation>::value;
-  static constexpr bool has_fast_atomics =
-      IsReduOptForFastAtomicFetch<T, BinaryOperation>::value;
-  static constexpr bool has_fast_reduce =
-      IsReduOptForFastReduce<T, BinaryOperation>::value;
-  static constexpr bool is_usm = IsUSM;
-  static constexpr bool is_placeholder =
-      (IsPlaceholder == access::placeholder::true_t);
-
-  // Only scalar (i.e. 0-dim and 1-dim with 1 element) reductions supported now.
-  // TODO: suport (Dims > 1) accessors/reductions.
-  // TODO: support true 1-Dimensional accessors/reductions (size() > 1).
-  // (size() == 1) is checked in the constructor of reduction_impl.
-  static_assert(Dims <= 1,
-                "Multi-dimensional reductions are not supported yet.");
+template <typename T, class BinaryOperation, int Dims, size_t Extent,
+          class Algorithm>
+class reduction_impl
+    : private reduction_impl_base,
+      public reduction_impl_algo<T, BinaryOperation, Dims, Extent, Algorithm> {
+private:
+  using algo = reduction_impl_algo<T, BinaryOperation, Dims, Extent, Algorithm>;
 
-  /// Returns the statically known identity value.
-  template <typename _T = T, class _BinaryOperation = BinaryOperation>
-  enable_if_t<IsKnownIdentityOp<_T, _BinaryOperation>::value,
-              _T> constexpr getIdentity() {
-    return reducer_type::getIdentity();
-  }
+public:
+  using reducer_type = typename algo::reducer_type;
+  // using result_type = T;
+  // using binary_operation = BinaryOperation;
```

Comment:
These supposed to be commented out? Are they simply there to relay that they have them from their superclass?

Reply:
Good catch, I meant to delete those.  Gone in 90deb60725f256b1b9343bc30ed21bf8ba3caffe.